ckcp/hack/util/destroy.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SCRIPT_DIR="$(
+
+  cd "$(dirname "$0")" >/dev/null || exit
+  pwd
+)"
+
+CONFIG="$SCRIPT_DIR/../../config.yaml"
+
+KUBECONFIG=${KUBECONFIG:-$HOME/.kube/config}
+
+# Removes different Argo CD applications deployed
+uninstall_pipeline_service(){
+    printf "\n Uninstalling Pipeline Service:\n"
+    # Remove Pipeline-Service application
+    kubectl delete -f "$SCRIPT_DIR/../../../gitops/argocd/argocd.yaml"
+    # Remove all the applications deployed by Pipeline-Service
+    kubectl delete -k "$SCRIPT_DIR/../../../gitops/argocd/argo-apps"
+}
+
+# Uninstalls operators
+uninstall_operators(){
+    # Openshift-Pipelines Operator
+    printf "\n Uninstalling Openshift-Pipelines Operator:\n"
```

Comment:
Yes, I've tried making the operator uninstallation more generic. 
There's still few exceptions in case of Pipelines Operator for e.g.- we've to start with uninstalling TektonConfig and then the rest of the resources related to Pipelines Operator.
So, my the common steps in uninstalling operators are - 
1. Remove the subscription.
2. Remove csv
3. Delete all the related CRDs
4. Delete the operator (doing this before deleting the CRDs would just respawn the operator).